src/dbally/views/freeform/text2sql/_view.py

```diff
@@ -91,22 +156,28 @@ async def ask(
 
     async def _generate_sql(
         self, query: str, conversation: PromptTemplate, llm_client: LLMClient, event_tracker: EventTracker
-    ) -> Tuple[str, PromptTemplate]:
+    ) -> Tuple[str, List[SQLParameterOption], PromptTemplate]:
         response = await llm_client.text_generation(
```

Comment:
How does the prompt builder fail? Generally, I believe that strings should be escaped when they are put into medium that needs the escaping, before that it should live in its canonical non-escaped form. So most likely it's prompt builder itself that should do the escaping - although its hard to tell, since I haven't seen the exact error message.

Reply:
I think this may be tricky to implement it inside prompt builder as it is not awere of what is placeholder and what is JSON

Reply:
![image](https://github.com/deepsense-ai/db-ally/assets/166019576/570b5f81-c331-46c3-999f-992f4e993971)

Reply:
Is the problem happening in db-ally's code or your code? Generally speaking prompt consists of two parts: prompt template (which is static) and dynamic values that are injected into placeholders defined in the template. The json is a dynamic value. The root of the problem seems to be that it is handled as part of the template, and not as a value injected into the template?